src/test/java/org/prebid/server/bidder/huaweiads/HuaweiAdSlotBuilderTest.java

```diff
@@ -633,7 +654,8 @@ public void buildShouldBuildNativeAdSlotWhenImpIsNativeAndHasTwoMainImagesAndOne
         final AdSlot30 expected = AdSlot30.builder()
                 .adType(3)
                 .slotId("slotId")
-                .detailedCreativeTypeList(List.of("903"))
+                .detailedCreativeTypeList(List.of("903", "901", "904", "905"))
+                .format(List.of())
```

Comment:
looks like some cases have become duplicates, it's better to re-write them in order to fully cover the following `foreach`:
```
.forEach(image -> {
                    if (numImage > 1
                            && HuaweiUtils.isFormatDefined(image.getW(), image.getH())
                            && HuaweiUtils.isFormatDefined(image.getWmin(), image.getHmin())) {
                        formats.add(Format.of(image.getW(), image.getH()));
                    }
                    if (numImage == 1
                            && HuaweiUtils.isFormatDefined(image.getW(), image.getH())
                            && HuaweiUtils.isFormatDefined(image.getWmin(), image.getHmin())) {
                        formats.addAll(filterPopularSizesByRatio(image.getW(), image.getH()));
                    }
                    if (numImage == 1
                            && !HuaweiUtils.isFormatDefined(image.getW(), image.getH())
                            && HuaweiUtils.isFormatDefined(image.getWmin(), image.getHmin())) {
                        formats.addAll(filterPopularSizesByRange(image.getWmin(), image.getHmin()));
                    }
                });
```

Reply:
Will do.